### ISSUE TITLE
Add toolchain management

### ISSUE THREAD

**Issue by cmr** [PistonDevelopers/VisualRust#172]
Currently, users are required to install and manage their own Rust toolchains. Let's make the user's life easier, and do it for them, along the lines of the [multirust](https://github.com/brson/multirust) script for Unix. Feature list:

- [ ] Select one of multiple toolchains for use (corresponding to Rust releases/release channels)
- [ ] Able to download and install/update toolchains
- [ ] Per-project toolchains
- [ ] Notifications to update with the ability to auto-update
- [ ] Support for both mingw and msvc targets

/cc @brson

**Comment by brson** [comment]
Yes! I think this would put VisualRust on a path to becoming *the* way to install Rust on Windows, making it similarly easy to Unix.

**Comment by brson** [comment]
FWIW I'm *intending* to switch my OS to Windows and may start finding time to contribute.

**Comment by brson** [comment]
It would though be very worthwhile to factor the release channel code out into something reusable for other tools.

**Comment by cmr** [comment]
@brson @alexcrichton what's the status/near-term viability of using only the msvc target?

**Comment by cmr** [comment]
(my understanding is that it's pretty solid)

**Comment by cmr** [comment]
Note that I'm explicitly ignoring cross-compiling for now. Can work that out in tandem with the official cross-compiling story.

**Comment by cmr** [comment]
I'm going to hardcode the PGP key and the HTTPS cert fingerprint, with the ability to override them somehow, to sidestep key management/trust for those.

**Comment by brson** [comment]
@cmr I believe @alexcrichton is optimistic that 64-bit will be ready to go for 1.3-1.4, but I'm not totally sure.

**Comment by alexcrichton** [comment]
A non-panicking pretty solid build is ready to go for 1.3, and I hope that we can sort out panicking support by 1.4 for 64-bit at least

**Comment by brson** [comment]
@Diggsey has a partial port of multirust to rust https://github.com/Diggsey/multirust-rs
